names/names.py

```diff
@@ -13,10 +14,39 @@
 duplicates = []  # Return the list of duplicates in this data structure
 
 # Replace the nested for loops below with your improvements
-for name_1 in names_1:
-    for name_2 in names_2:
-        if name_1 == name_2:
-            duplicates.append(name_1)
+# ORIGINAL CODE
+# RUNTIME O(n**2)
+
+# for name_1 in names_1:
+#     for name_2 in names_2:
+#         if name_1 == name_2:
+#             duplicates.append(name_1)
+
+# FASTER SOLUTION, but I need to use a data structure :facepalm:, almost did not see that
+# for name_1 in names_1:
+#     if name_1 in names_2:
+#         print('name', name_1)
+#         duplicates.append(name_1)
+
+# BST solution
+new_tree = BinarySearchTree(names_1[0])
+
+for name in range(1, len(names_1)):
+    new_tree.insert(names_1[name])
+
+for name in names_2:
+    if new_tree.contains(name):
+        duplicates.append(name)
+
+# could not figure out how to compare to BSTs :(
```

Comment:
I'm not sure that comparing two BST's would have made this run any faster than your implementation, I could be wrong, but I believe your solution is the most optimal. So nice job!!